src/main/java/it/gov/pagopa/pu/sil/connector/auth/service/AuthnService.java

```diff
@@ -0,0 +1,9 @@
+package it.gov.pagopa.pu.sil.connector.auth.service;
+
+import it.gov.pagopa.pu.auth.dto.generated.AccessToken;
+
+public interface AuthnService {
+  AccessToken postToken(String clientId, String grantType, String scope,
```

Comment:
let's call it, and just ask for the ipaCode... let's hide all the details on how to make the M2M auth on this connector service class
```suggestion
  AccessToken getM2MToken(
```